Terminal.Gui/Views/TileView.cs

```diff
@@ -111,17 +189,26 @@ protected virtual void OnSplitterMoved (int idx)
 		}
 
 		/// <summary>
-		/// Scraps all <see cref="Tiles"/>  and creates <paramref name="count"/> new tiles
+		/// Scraps all <see cref="Tiles"/> and creates <paramref name="count"/> new tiles
 		/// in orientation <see cref="Orientation"/>
 		/// </summary>
 		/// <param name="count"></param>
 		public void RebuildForTileCount (int count)
 		{
 			tiles = new List<Tile> ();
 			splitterDistances = new List<Pos> ();
+			if (splitterLines != null) {
+				foreach (var sl in splitterLines) {
+					sl.Dispose ();
+				}
+			}
 			splitterLines = new List<TileViewLineView> ();
 
 			RemoveAll ();
+			foreach (var tile in tiles) {
+				tile.ContentView.Dispose ();
```

Comment:
Disposing a `View` disposes the whole tree which could be a problem if the user maintains references to stuff in tiles and wants to put them back in the new configuration.

For example the following use case?

```
// Create TextView

// Add to Tile

// Do some stuff

// Rebuild for Tile count +1

// Add TextView back in at Tile element 0 (user won't know but TextView has actually already been disposed)
```

Reply:
Nevermind think I am panicing about nothing.  Just wrote this test and it passes

```csharp
[Fact,AutoInitShutdown]
public void TestDisposal()
{
	var tv = GetTileView (20,10);

	var myReusableView = new DisposeCounter ();

	// I want my view in the first tile
	tv.Tiles.ElementAt (0).ContentView.Add (myReusableView);
	Assert.Equal (0, myReusableView.DisposalCount);

	// I've changed my mind, I want 3 tiles now
	tv.RebuildForTileCount (3);

	// but I still want my view in the first tile
	tv.Tiles.ElementAt (0).ContentView.Add (myReusableView);
	Assert.Multiple (
		()=>Assert.Equal (0, myReusableView.DisposalCount)
		,()=> {
			tv.Dispose ();
			Assert.Equal (1, myReusableView.DisposalCount); 
		});
}
```

Reply:
Actually `tiles` is always empty at this point because line 198 initializes it to a new list.